SCAutolib/models/gui.py

```diff
@@ -0,0 +1,296 @@
+from SCAutolib import run
+
+import os
+from time import sleep, time
+import uinput
+import cv2
+import numpy as np
+import pandas as pd
+from io import StringIO
+import pytesseract
+import keyboard
+import uinput
+import functools
+
+
+class Screen():
+    '''Captures the screenshots'''
+
+    def __init__(self, directory):
+        self.directory=directory
+        self.screenshot_num = 1
+
+    def screenshot(self):
+        '''
+        Runs ffmpeg to take a screenshot.
+
+        The filename of screenshot is then returned.
+        '''
+
+        filename = f'/tmp/SC-tests/{self.screenshot_num}.png'
+        out = run(['ffmpeg', '-hide_banner', '-y', '-f', 'kmsgrab', '-i', '-',
+             '-vf', 'hwdownload,format=bgr0', '-frames', '1', '-update', '1',
+             filename
+        ])
+
+        self.screenshot_num += 1
+        return filename
+
+
+class Mouse():
+    def __init__(self):
+        run(['modprobe', 'uinput'], check=True)
+        # initialize the uinput device
+        self.device = uinput.Device([
+            uinput.REL_X,
+            uinput.REL_Y,
+            uinput.BTN_LEFT,
+            uinput.BTN_MIDDLE,
+            uinput.BTN_RIGHT,
+            uinput.REL_WHEEL,
+        ])
+
+    def move(self, x, y):
+        '''Moves the mouse cursor to specified absolute coordinate.'''
+
+        # Go to upper left corner
+        self.device.emit(uinput.REL_X, -10000)
+        sleep(0.1)
+        self.device.emit(uinput.REL_Y, -10000)
+        sleep(0.1)
+
+        # Move to correct X coordinate
+        for i in range(x):
+            self.device.emit(uinput.REL_X, 1)
+            sleep(0.005)
+
+        # Move to correct Y coordinate
+        for i in range(y):
+            self.device.emit(uinput.REL_Y, 1)
+            sleep(0.005)
```

Comment:
I see certen pattern using of `.emit` method
```python
self.device.emit(some_argumnet, number)
sleep(some_seconds)
```
So, I would suggest to extract this to small internalmethod with relevant parameters and use it